### ISSUE TITLE
[Request] QModal to support Array syntax for content-css and content-classes

### ISSUE THREAD

**Issue by daniel-shuy** [quasarframework/quasar#856]
### Software version

Quasar: 0.14.2

### What did you get as the error?
```
[Vue warn]: Invalid prop: type check failed for prop "contentClasses". Expected Object, String, got Array.

found in

---> <QModal>
```
### What were you expecting?

From the Vue guide (https://vuejs.org/v2/guide/class-and-style.html#Array-Syntax), Class/Style Bindings can accept Arrays.  One would expect the same behavior for Quasar components.

I'm not sure if this also applies to other Quasar components, but this is the only one I've noticed so far.

### What steps did you take, to get the error?

Example:
``` vue
<q-modal :content-classes="[myClass1, myClass2]">
  <!-- ... -->
</q-modal>
```

**Comment by rstoenescu** [comment]
The documentation for QModal clearly states the content-classes needs an Object only. This is due to performance under the covers. Making it work with all types would have a small perf penalty. Will ponder on this.

**Comment by daniel-shuy** [comment]
I think supporting the Array syntax is more important than the Object syntax though.

It is possible to nest the Object syntax in the Array syntax, eg.
``` vue
<q-modal :content-classes="[{hidden: isHidden}, myClass]">
  <!-- ... -->
</q-modal>
```

However trying to do the reverse (nesting Array syntax in Object syntax) is really ugly and cumbersome:
``` vue
<q-modal :content-classes="{myClass1: true, myClass2: true}">
  <!-- ... -->
</q-modal>
```

If the prop type validation is causing the performance penalty, what about removing type validation for `contentClasses`/`contentCss` altogether? Vue itself doesn't seem to validate the `class`/`style` props anyways.

**Comment by rstoenescu** [comment]
Agreed. Will do. I was just thinking of the overhead implied in parsing all the possibilities.

**Comment by rstoenescu** [comment]
Available in edge and future v0.14.3
